zerver/tests/test_event_queue.py

```diff
@@ -27,6 +27,20 @@ class MissedMessageNotificationsTest(ZulipTestCase):
     """Tests the logic for when missed-message notifications
     should be triggered, based on user settings"""
 
+    def _send_stream_message(
+        self,
+        sender: UserProfile,
+        stream_name: str,
+        content: str = "test content",
+        topic_name: str = "test",
+    ) -> int:
+        with self.captureOnCommitCallbacks(execute=True):
+            return self.send_stream_message(sender, stream_name, content, topic_name)
+
+    def _send_personal_message(self, from_user: UserProfile, to_user: UserProfile) -> int:
+        with self.captureOnCommitCallbacks(execute=True):
+            return self.send_personal_message(from_user, to_user)
```

Comment:
By default, the `on_commit` callbacks are not executed in tests, so if we always use `captureOnCommitCallbacks` in `send_stream_message` then the events will be sent everytime when calling `send_stream_message` and we would need to update many other tests to handle this.

But, yeah I agree we can define a separate function for it in `test_classes.py` itself and use that directly in all the files instead of defining it again and again in separate files.

Reply:
Just to toss one idea out, we could make it a parameter to `send_personal_message`, rather than a separate function, and use a [conditional `with`](https://stackoverflow.com/a/34798330).